src/ia-book-search-results.js

```diff
@@ -81,6 +100,17 @@ export class IABookSearchResults extends LitElement {
     return this.renderSearchAllFiles ? controls : nothing;
   }
 
+  get loadingIndicator() {
+    const loadingTemplate = html`
+      <div class="loading">
+        <ia-activity-indicator mode="processing"></ia-activity-indicator>
+        <p>Searching</p>
+        <button @click=${this.cancelSearch}>Cancel</button>
+      </div>
+    `;
+    return this.queryInProgress ? loadingTemplate : nothing;
```

Comment:
_Suggestion_ Separate the loading indicator template from the logic to display it.

When you read the calling code below `${this.loadingIndicator}`, it looks like you're putting the loading indicator there, but it will only show up when certain conditions are met internally.

That condition becomes more clear if you put it inline, ie:
```
${this.queryInProgress ? this.loadingIndicator : nothing}
```

Reply:
You know, I originally wrote it in that pattern, but realized the `headerSection` template getter had the ternary in the getter, and when I last updated this component to conditionally render the search multiple template, I followed that convention. My instinct is to put the logic in the render method, however at this point all of these submenu components have `headerSection` getters that moved the logic into the getter. I think in the interest of getting this out as a beta test this week, we'll have to refactor this and the other components to match later on.